src/python/qeqiskit/backend/backend.py

```diff
@@ -276,6 +281,33 @@ def run_circuitset_and_measure(
 
         return self.aggregregate_measurements(jobs, batches, multiplicities)
 
+    def execute_with_retries(
```

Comment:
I would add an optional timeout here?

Reply:
Added a timeout in 547efefbaae0e3b639ba1a22c586414c66e2bbb0. Unfortunately it's hard to test the behavior without a mock server because the IBMQ servers sometimes the jobs finish really quickly. I left some comments about this in the tests.